tests/test_problems.py

```diff
@@ -869,9 +876,62 @@ def test_problem_instance(self):
             "generic_qubo":['problem_type']
         }
 
-        for k,v in all_qubos.items():
+        for k,v in qubos.items():
             assert list(v.problem_instance.keys()) == expected_keys[k], "Problem instance keys are not correct for problem type {}".format(k)
             assert k == v.problem_instance['problem_type'], "Problem type is not correct for problem type {}".format(k)
 
+
+    def test_problem_from_instance_dict(self):
+        """
+        Test problem from instance method of the problem class.
+        """
+        problem_mapper = {
+            "generic_qubo": QUBO,
+            "tsp": TSP,
+            "number_partition": NumberPartition,
+            "maximum_cut": MaximumCut,
+            "knapsack": Knapsack,
+            "slack_free_knapsack": SlackFreeKnapsack,
+            "minimum_vertex_cover": MinimumVertexCover,
+            "shortest_path": ShortestPath,
+        }
+
+        problems, qubos = self.__generate_random_problems()
+
+        for type in qubos:
+            if type == "generic_qubo":
+                continue
+
+            problem_instance = qubos[type].problem_instance.copy()
+
+            problem = Problem.from_instance_dict(problem_instance)
+
+            assert problem.problem_instance == problems[type].problem_instance, "Problem from instance method is not correct for problem type {}".format(type)
+            assert convert2serialize(problem) == convert2serialize(problems[type]), "Problem from instance method is not correct for problem type {}".format(type)
+
+        
+    def test_qubo_from_dict(self):
+        """
+        Test qubo from dict method of the QUBO class.
+        """
+
+        _, qubos = self.__generate_random_problems()
+        for _, qubo in qubos.items():
+
+            qubo_dict = qubo.asdict()
+
+            new_qubo = QUBO.from_dict(qubo_dict)
```

Comment:
I think we also need to check the constant term here: in the older code we have at times forgotten to take properly care of it